### PR TITLE
[zelos] Set a maximum width for dynamic connection shapes.

### PR DESCRIPTION
**Pull request by samelhusseini**
## The basics
- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Fixes https://github.com/google/blockly/issues/3684 and https://github.com/google/blockly/issues/3676

### Proposed Changes

Set a maximum for the width of a dynamic connection shape, thus getting to a shape like this: 

Screen Shot 2020-02-07 at 1 29 16 PM
and
Screen Shot 2020-02-07 at 1 34 28 PM

For dealing with the statement input + output connection. We keep track if we have any statement inputs in our input rows, and only render the left side of the connection if that is the case, like so: 

Screen Shot 2020-02-07 at 1 46 29 PM or 
Screen Shot 2020-02-07 at 1 47 39 PM

### Reason for Changes

Size of multi-line blocks with an output connection grows without a limit creating obnoxiously sized blocks.

### Test Coverage

Tested in playground.

Tested on:

### Additional Information

As a side effect of this change, when blocks with output connections are deeply nested, at some point (after hitting the max width), the blocks don't visually align exactly. I think that's a fair side effect. 

Thoughts @rachel-fenichel?

Screen Shot 2020-02-07 at 1 15 25 PM

### CODE REVIEW

File: core/renderers/zelos/constants.js [hunk 1]
```diff
@@ -492,22 +503,31 @@ Blockly.zelos.ConstantProvider.prototype.makeHexagonal = function() {
  * @package
  */
 Blockly.zelos.ConstantProvider.prototype.makeRounded = function() {
+  var maxWidth = this.MAX_DYNAMIC_CONNECTION_SHAPE_WIDTH;
+  var maxHeight = maxWidth * 2;
+
   // The main path for the rounded connection shape is made out of a single arc.
```
Comment (alschmiedt) [inline core/renderers/zelos/constants.js]: Update comment

File: core/renderers/common/info.js [hunk 2]
```diff
@@ -139,11 +139,10 @@ Blockly.blockRendering.RenderInfo = function(renderer, block) {
...
 
   /**
-   * The total number of input rows added onto the block.
-   * @type {number}
-   * @protected
+   * An array of input rows on the block.
+   * @type {!Array.<!Blockly.blockRendering.InputRow>}
    */
-  this.inputRowNum_ = 1;
+  this.inputRows = [];
```
Comment (alschmiedt) [inline core/renderers/common/info.js]: Just curious why the change from just storing the number of inputs?

Comment (samelhusseini) [reply core/renderers/common/info.js]: I think that way we can update our "not so good looking" for loops from looping through all the rows to just the input rows.
